docs/static/event-data.asciidoc

```diff
@@ -0,0 +1,417 @@
+
+Make use of the `@metadata` field any time you need a temporary field but do not
+want it to be in the final output.
+
+Perhaps one of the most common use cases for this new field is with the `date`
+filter and having a temporary timestamp.
+
+This configuration file has been simplified, but uses the timestamp format
+common to Apache and Nginx web servers.  In the past, you'd have to delete
+the timestamp field yourself, after using it to overwrite the `@timestamp`
+field.  With the `@metadata` field, this is no longer necessary:
+
+[source,ruby]
+----------------------------------
+input { stdin { } }
+
+filter {
+  grok { match => [ "message", "%{HTTPDATE:[@metadata][timestamp]}" ] }
+  date { match => [ "[@metadata][timestamp]", "dd/MMM/yyyy:HH:mm:ss Z" ] }
+}
+
+output {
+  stdout { codec => rubydebug }
+}
+----------------------------------
+
+Notice that this configuration puts the extracted date into the
+`[@metadata][timestamp]` field in the `grok` filter.  Let's feed this
+configuration a sample date string and see what comes out:
+
+[source,ruby]
+----------------------------------
+$ bin/logstash -f ../test.conf
+Pipeline main started
+02/Mar/2014:15:36:43 +0100
+{
+    "@timestamp" => 2014-03-02T14:36:43.000Z,
+      "@version" => "1",
+          "host" => "example.com",
+       "message" => "02/Mar/2014:15:36:43 +0100"
+}
+----------------------------------
+
+That's it!  No extra fields in the output, and a cleaner config file because you
+do not have to delete a "timestamp" field after conversion in the `date` filter.
+
+Another use case is the https://github.com/logstash-plugins/logstash-input-couchdb_changes[CouchDB Changes input plugin]. 
+This plugin automatically captures CouchDB document field metadata into the
+`@metadata` field within the input plugin itself.  When the events pass through
+to be indexed by Elasticsearch, the Elasticsearch output plugin allows you to
+specify the `action` (delete, update, insert, etc.) and the `document_id`, like
+this:
+
+[source,ruby]
+----------------------------------
+output {
+  elasticsearch {
+    action => "%{[@metadata][action]}"
+    document_id => "%{[@metadata][_id]}"
+    hosts => ["example.com"]
+    index => "index_name"
+    protocol => "http"
+  }
+}
+----------------------------------
+
+[discrete]
+[[date-time]]
+===== sprintf date/time format in conditionals
+
+Sprintf date/time format in conditionals is not currently supported, but a workaround is available. 
+Put the date calculation in a field so that you can use the field reference in a conditional. 
+
+*Example* 
+
+Using sprintf time format directly to add a field based on ingestion time _will not work_: 
+// This counter example is formatted to be understated to help users avoid following a bad example 
+
+```
+----------
+# non-working example
+filter{
+  if "%{+HH}:%{+mm}" < "16:30" {
+    mutate {
+      add_field => { "string_compare" => "%{+HH}:%{+mm} is before 16:30" }
+    }
+  }
+}
+----------
+```
```

Comment:
```suggestion
[source,ruby]
----------
# non-working example
filter{
  if "%{+HH}:%{+mm}" < "16:30" {
    mutate {
      add_field => { "string_compare" => "%{+HH}:%{+mm} is before 16:30" }
    }
  }
}
----------
```

Reply:
I formatted this non-working example to be understated rather than bold like true examples, with the intention of emphasizing the good example that follows.  (See note at line 384.)  

This PR restructures files, with content changes coming in a soon-to-be follow up PR. We can evaluate this comment for inclusion in that PR.

Reply:
Noted.